File: packages/common/src/utils/chainUuidV5.ts
import {v5 as uuidv5} from 'uuid';

export function chainUuidV5(namespaceUuid: string, ...values: [string, ...string[]]): string {
  let result = namespaceUuid;
  for (const value of values) {
    result = uuidv5(value, result);
  }
  return result;
}

File: lib/chainUUIDV5.go
package lib

import "encore.dev/types/uuid"

func ChainUUIDV5(namespace uuid.UUID, subjects ...string) uuid.UUID {
	if len(subjects) == 0 {
		panic("chainUUIDV5: no subjects provided")
	}

	final := namespace
	for _, subject := range subjects {
		final = uuid.NewV5(final, subject)
	}
	return final
}
